### PR TITLE
speed: Also measure RSA encrypt/decrypt, not only RSA sign/verify

### PR DESCRIPTION
**Pull request by ifranzki**
While RSA encrypt/decrypt and sign/verify are basically the same mod-expo operations, the speed of the operation may still differ, due to different padding, as well as the use of implicit rejection for RSA decrypt.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: apps/speed.c
Comment (t8m) [inline apps/speed.c:2900]: I do not think the apostrophe in `RSA sign's`  is right. Similarly in the below cases.

Comment (tom-cosgrove-arm) [reply apps/speed.c:2900]: Hmm, I'm going to argue devil's advocate here:

An apostrophe is to stand in for missing letters (e.g. `it is` -> `its`) or for possessive (`Tom's bicycle`, although not in `his`, `hers`, `its`)

In this case, the full phrase would be `xxx bits private RSA signatures in ...` so the apostrophe _is_ standing in for missing letters!

But I agree, it reads horribly, and I would prefer something else here too

Comment (ifranzki) [reply apps/speed.c:2900]: So to what do you want me to change it? 
Just no apostrophe? Or spelled out `RSA signatures`, `RSA verifications`, `RSA encryptions`, `RSA decryptions` ?

Comment (tom-cosgrove-arm) [reply apps/speed.c:2900]: Well, TBH the previous `RSA's` isn't correct, although it is "less wrong" ? or at least seen enough that it could be considered "acceptable by some" to use an apostrophe for plural of an acronym like this.

For signatures, `N M-bit RSA signs` would match what is done for DSA, ECDSA, etc, and seems fine.

Those other algs use singular `verify` which makes me shudder when I read it!

Now, it's also common to abbreviate with a dot (period)  (most often with things such as `e.g.`) - so we could do something like

```
Doing 2048-bit private RSA sigs. for 10s: 17637 2048-bit RSA sigs. in 9.99s
Doing 2048-bit public RSA verifs. for 10s: 712226 2048-bit RSA verifs. in 9.99s
```

The other question of course is: could anyone be consuming this output by script in a CI, and would changes here break that? (Actually @zorrorffm do _we_ do that?)

Comment (ifranzki) [reply apps/speed.c:2900]: How about `xxx bits private RSA sign ops in ...` ? So the plural would be on `operations `(abbreviated as `ops`), not on sign, verify, etc.....

Comment (zorrorffm) [reply apps/speed.c:2900]: > The other question of course is: could anyone be consuming this output by script in a CI, and would changes here break that? (Actually @zorrorffm do _we_ do that?)

Yes, our CI consumes the final output, the changes at line 4101 will break it. Thanks for the reminder, I will change our CI accordingly after this PR is merged.

Comment (ifranzki) [reply apps/speed.c:2900]: So I can easily change the code to use `xxx bits private RSA sign ops in ...` instead, but the question is: Only for RSA? That would be somewhat inconsistent with the other algos. Or should I change it for all algos to use `...ops` ? That would be a slightly larger change (and not really related to the topic of this PR). I guess I would then do this in a 2nd commit. 

Let me know what you think and if I should change it to `... ops` and where.

Comment (ifranzki) [reply apps/speed.c:2900]: @t8m @paulidale any opinion about how to continue with this?

Comment (tom-cosgrove-arm) [reply apps/speed.c:2900]: I am happy with `sign ops` and `verify ops` with no punctuation

Comment (paulidale) [reply apps/speed.c:2900]: Let's go with @tom-cosgrove-arm's suggestion.

Comment (ifranzki) [reply apps/speed.c:2900]: OK, and just for the RSA places that I touch anyway, or for all algos ?

Comment (tom-cosgrove-arm) [reply apps/speed.c:2900]: I think if we're going to change output, let's be consistent, and let's do it in one go (if it breaks something, better for folks to just have to fixup once)

So I think yes, all algs

Comment (t8m) [reply apps/speed.c:2900]: > OK, and just for the RSA places that I touch anyway, or for all algos ?

I would leave that up to you

**Comment by ifranzki** [comment]
Changed my commit to use `sign ops`, `verify ops`, etc. as discussed.

Also changed the other algos to use this notation in a 2nd commit. Just in case you don't like that, we can easily drop this commit :-)

**Review by paulidale — APPROVED** [review]

**Comment by ifranzki** [comment]
@t8m friendly ping .....

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**
